components/cors-mgt/org.wso2.carbon.identity.cors.mgt.core/src/main/java/org/wso2/carbon/identity/cors/mgt/core/internal/impl/CORSManagementServiceImpl.java

```diff
@@ -217,44 +374,31 @@ private void validateTenantDomain(String tenantDomain) throws CORSManagementServ
     }
 
     /**
-     * Validate the CORSOrigin list.
+     * Validate the application ID.
      *
-     * @param corsOrigins List of CORSOrigin instances.
-     * @throws CORSManagementServiceClientException
+     * @param tenantDomain The tenant domain.
+     * @param appId        The application ID.
      */
-    private void validateOrigins(List<CORSOrigin> corsOrigins) throws CORSManagementServiceClientException {
+    private void validateApplicationId(String tenantDomain, String appId) throws CORSManagementServiceClientException {
 
-        if (corsOrigins == null) {
-            if (log.isDebugEnabled()) {
-                log.debug(ERROR_CODE_EMPTY_LIST.getMessage());
-            }
-            throw handleClientException(ERROR_CODE_EMPTY_LIST);
+        // If the appId is blank then the CORS origins that should be set/add/delete belong in the tenant level.
+        if (StringUtils.isBlank(appId)) {
+            return;
         }
 
-        for (CORSOrigin corsOrigin : corsOrigins) {
-            if (isInvalidOrigin(corsOrigin)) {
-                if (log.isDebugEnabled()) {
-                    log.debug(String.format(ERROR_CODE_INVALID_ORIGIN.getMessage(), corsOrigin.getUrl()));
-                }
-                throw handleClientException(ERROR_CODE_INVALID_ORIGIN, corsOrigin.getUrl());
-            }
-        }
-    }
-
-    /**
-     * Check if the format of the Origin is valid.
-     *
-     * @param origin Origin to be checked for validity.
-     * @return {@code true} if the origin is valid, {@code false} otherwise.
-     */
-    private boolean isInvalidOrigin(CORSOrigin origin) {
-
+        // Check whether the appId belongs to the tenant with the tenantDomain.
         try {
-            new URL(origin.getUrl()).toURI();
-        } catch (MalformedURLException | URISyntaxException e) {
-            return true;
+            ApplicationBasicInfo applicationBasicInfo = ApplicationManagementService.getInstance()
+                    .getApplicationBasicInfoByResourceId(appId, tenantDomain);
+            if (applicationBasicInfo == null) {
+                throw handleClientException(ErrorMessages.ERROR_CODE_INVALID_APP_ID, appId);
+            }
+        } catch (IdentityApplicationManagementException e) {
+            // Something else happened.
+            if (log.isDebugEnabled()) {
+                log.debug(e);
```

Comment:
add error message with contextual information

Reply:
The error already has a message.